source/common/upstream/load_stats_reporter.cc

```diff
@@ -106,27 +111,36 @@ void LoadStatsReporter::onReceiveInitialMetadata(Http::HeaderMapPtr&& metadata)
 void LoadStatsReporter::onReceiveMessage(
     std::unique_ptr<envoy::api::v2::LoadStatsResponse>&& message) {
   ENVOY_LOG(debug, "New load report epoch: {}", message->DebugString());
+  stats_.requests_.inc();
+  message_ = std::move(message);
+  prepareForResponse();
+}
+
+void LoadStatsReporter::prepareForResponse() {
```

Comment:
This would be clearer as `startLoadReportPeriod` or something, since there is an unfortunate inversion of the usual request/response paradigm here, where a response is actually a `LoadStatsRequest`.